examples/test_util/EclIO.cpp

```diff
@@ -0,0 +1,1179 @@
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include "EclIO.hpp"
+
+#include <iterator>
+#include <iomanip>
+#include <stdexcept>
+
+bool EclIO::isEOF(std::fstream *fileH){
+    
+    /* try to read an 4 byte integer, if eof of file passed return true.
+       if not eof, seek back to current file possition  */
+    
+    int num;
+    long int pos=fileH->tellg();
+    fileH->read((char *)&num, sizeof(num));
+    
+    if (fileH->eof()){
+        return true;
+    } else {
+        fileH->seekg (pos);
+        return false;
+    }
+}
+
+int EclIO::reverseInt(int num){
+    
+    int b0,b1,b2,b3;
+
+    b0= (num & 0x000000FF)>>0;
+    b1= (num & 0x0000FF00)>>8;
+    b2= (num & 0x00FF0000)>>16;
+    b3= (num & 0xFF000000)>>24;
+
+    num= (b0<<24) | (b1<<16) | (b2<<8) | (b3<<0) ;
+
+    return num;
+}
+
+float EclIO::reverseFloat(float num){
+    
+    float retVal;
+    char *floatToConvert = ( char* ) & num;
+    char *returnFloat = ( char* ) & retVal;
+
+    // swap the bytes into a temporary buffer
+    returnFloat[0] = floatToConvert[3];
+    returnFloat[1] = floatToConvert[2];
+    returnFloat[2] = floatToConvert[1];
+    returnFloat[3] = floatToConvert[0];
+
+    return retVal;
+}
+
+double EclIO::reverseDouble(const double num)
+{
+    double retVal;
+    char *doubleToConvert = ( char* ) & num;
+    char *returnDouble = ( char* ) & retVal;
+
+    // swap the bytes into a temporary buffer
+
+    returnDouble[0] = doubleToConvert[7];
+    returnDouble[1] = doubleToConvert[6];
+    returnDouble[2] = doubleToConvert[5];
+    returnDouble[3] = doubleToConvert[4];
+    returnDouble[4] = doubleToConvert[3];
+    returnDouble[5] = doubleToConvert[2];
+    returnDouble[6] = doubleToConvert[1];
+    returnDouble[7] = doubleToConvert[0];
+
+    return retVal;
+}   
+
+std::vector<std::string> EclIO::split_string(const std::string &instr){
+
+    std::istringstream iss(instr);
+
+    std::vector<std::string> tokens{std::istream_iterator<std::string>{iss},
+        std::istream_iterator<std::string>{}};     
+ 
+    return tokens;
+}
+
+
+template <class T> std::string EclIO::make_scientific_string(T number){
+
+    std::stringstream buffer;
+    std::string str1,str2,exStr;
+    int p1,exp;
+    bool isDouble=false;
+    
+    if (sizeof(number)==8){
+        isDouble=true;
+    }
+    
+    if (isDouble==true){
```

Comment:
If you need to specialize you should use 
```
template<>
std::string EclIO::make_scientific_string(double number)
```
not specialize on the type size - i.e. here you would get the 'D' if you passed a `std::size_t` on a 64 bit platform.

Reply:
Throw exception if T is not float or double. Using function typeid to check if float or double (instead of size)

    size_t precision = typeid(number)==typeid(float) ? 7 : 13;